core/src/v2/models.rs

```diff
@@ -825,6 +824,13 @@ where
             Resolvable::Resolved { ref old, .. } => old.read().description().map(String::from),
         }
     }
+
+    pub fn read(&self) -> parking_lot::RwLockReadGuard<S> {
+        match self {
+            Resolvable::Raw(s) => s.read(),
+            Resolvable::Resolved { old, .. } => old.read(),
```

Comment:
I think the failure is coming from this line. We should read the resolved schema, not the unresolved one.

Reply:
I also wonder what's the motivation behind this addition. This type already implements `Deref` to `Arc<RwLock<S>>`, so we won't really need this additional method.

Reply:
Oh, yes, thanks.
I added this function by accident, will remove.